pkg/kube/cluster-init.sh

```diff
@@ -54,6 +54,44 @@ wait_for_default_route() {
   return 1
 }
 
+wait_for_vault() {
+        # Wait for vault to unseal
+        vaultMgrStatusPath="/run/vaultmgr/VaultStatus/Application Data Store.json"
+        vaultMgrStatus=0
+        DataSecAtRestStatus_DATASEC_AT_REST_DISABLED=1
+        DataSecAtRestStatus_DATASEC_AT_REST_ENABLED=2
+        while [ $vaultMgrStatus -ne $DataSecAtRestStatus_DATASEC_AT_REST_DISABLED ] && [ $vaultMgrStatus -ne $DataSecAtRestStatus_DATASEC_AT_REST_ENABLED ];
+        do
+                logmsg "Waiting for vault, currently: $vaultMgrStatus"
+                if [ -e "$vaultMgrStatusPath" ]; then
+                        vaultMgrStatus=$(jq -r .Status "$vaultMgrStatusPath")
+                fi
+                if [ "$vaultMgrStatus" -ne "$DataSecAtRestStatus_DATASEC_AT_REST_DISABLED" ] && [ "$vaultMgrStatus" -ne "$DataSecAtRestStatus_DATASEC_AT_REST_ENABLED" ]; then
+                        sleep 1
+                fi
+        done
+}
+
+mount_etcd_vol() {
+        # NOTE: We only support zfs storage in production systems because data is persisted on zvol.
+        etcd_fs_ready=0
+        etcd_dev="/dev/zvol/persist/etcd-storage"
+        logmsg "Waiting for $etcd_dev"
+        while [ $etcd_fs_ready -eq 0 ];
+        do
+                sleep 1
```

Comment:
Do you always want to sleep 1, or only if not yet ready?

Reply:
Sure I'll clean that up

Reply:
This fs check was redundant since the fs is setup in vaultmgr.  Condensed it and only sleep if block device not available now.